# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/CamerasTypes.h
# Commit: 61ca6f60af1a
# Full Hash: 61ca6f60af1aabccfb8693c060e460cfe665ae0e
# Author: Jan Grulich <jgrulich@redhat.com>
# Date: 2024-02-01 17:05:48
# Description:
#   Bug 1877679 - Fix serialization of CameraAccessStatus enum r=pehrsons
#   
#   Use ContiguousEnumSerializerInclusive providing highest legal value
#   instead as the CameraAccessStatus enum doesn't have a sentinel value.
#   
# ==============================================================================

diff -r 93128c0fb8fe -r 61ca6f60af1a dom/media/systemservices/CamerasTypes.h
--- a/dom/media/systemservices/CamerasTypes.h	Thu Feb 01 09:57:19 2024 +0200
+++ b/dom/media/systemservices/CamerasTypes.h	Thu Feb 01 08:03:58 2024 +0000
@@ -52,7 +52,7 @@
 
 template <>
 struct ParamTraits<mozilla::camera::CamerasAccessStatus>
-    : public ContiguousEnumSerializer<
+    : public ContiguousEnumSerializerInclusive<
           mozilla::camera::CamerasAccessStatus,
           mozilla::camera::CamerasAccessStatus::Granted,
           mozilla::camera::CamerasAccessStatus::Error> {};
